### PR TITLE
chore: update copyright and spdx

### PR DESCRIPTION
**Pull request by tenthirtyam**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/terraform-provider-vcf/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

As discussed with Chip Childers:

- Updates or adds the Broadcom copyright, as needed.
- Updates or adds the SPDX-License-Identifier, as needed.

```go
// © Broadcom. All Rights Reserved.
// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
// SPDX-License-Identifier: MPL-2.0
```

Note: There is an additioanl update to `internal/provider/provider.go` to ensure `ResourcesMap` and `DataSourcesMap` entries are in **alphabetical order** for readability.

**Type of Pull Request**

- [ ] This is a bug fix.
- [ ] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [x] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

**Test and Documentation Coverage**

For bug fixes or features:

- [x] Tests have been completed.
- [x] Documentation has been added/updated.

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -108,7 +109,7 @@ func providerConfigure(_ context.Context, data *schema.ResourceData) (interface{
 		password, isSetPassword := data.GetOk("sddc_manager_password")
 		hostName, isSetHost := data.GetOk("sddc_manager_host")
 		if !isVcfUsernameSet || !isSetPassword || !isSetHost {
-			return nil, diag.Errorf("SDDC Manager username, password and host must be provided")
+			return nil, diag.Errorf("error: SDDC Manager username, password, and host must be provided")
```
Comment (dimitarproynov) [inline internal/provider/provider.go:112]: "error: " is redundant as a Error struct is returned which the terraform SDK will log at error level.

File: internal/provider/provider.go [hunk 2]
```diff
@@ -122,7 +123,7 @@ func providerConfigure(_ context.Context, data *schema.ResourceData) (interface{
 		password, isSetPassword := data.GetOk("cloud_builder_password")
 		hostName, isSetHost := data.GetOk("cloud_builder_host")
 		if !isCbUsernameSet || !isSetPassword || !isSetHost {
-			return nil, diag.Errorf("CloudBuilder username, password and host must be provided")
+			return nil, diag.Errorf("error: Cloud Builder username, password, and host must be provided")
```
Comment (dimitarproynov) [inline internal/provider/provider.go:126]: same as above

**Review by dimitarproynov — DISMISSED** [review]

File: internal/provider/provider.go (hunk 1 above)
Comment (tenthirtyam) [reply internal/provider/provider.go]: Resolved in e8f6082.

File: internal/provider/provider.go (hunk 2 above)
Comment (tenthirtyam) [reply internal/provider/provider.go]: Resolved in e8f6082.

**Pull request merged**